utils.py

```diff
@@ -70,7 +73,7 @@ def full_lora_path_for(lora: str) -> Optional[str]:
         lora += ".safetensors"
 
     # Find the matching lora path
-    matching_lora = next((x for x in folder_paths.get_filename_list("loras") if x.endswith(lora)), None)
+    matching_lora = next((x for x in folder_paths.get_filename_list("loras") if Path(x).name == lora), None)
```

Comment:
Let's do it also for the LoRAs.
```suggestion
    matching_lora = next((x for x in folder_paths.get_filename_list("loras") if Path(x).name == lora), None)
    matching_lora = (matching_lora if matching_lora is not None else
                next((x for x in folder_paths.get_filename_list("loras") if Path(x).stem == Path(lora).stem), None))
```

Reply:
Also, please avoid overly long lines/expressions. Something like this would be easier to read:
```py
matching_lora = next((x for x in folder_paths.get_filename_list("loras") if Path(x).name == lora), None)
if matching_lora is None:
    matching_lora = next((x for x in folder_paths.get_filename_list("loras") if Path(x).stem == lora), None)  
```